app/api/utils.py

```diff
@@ -479,3 +474,29 @@ def readinto(self, b):
                 return 0    # indicate EOF
 
     return io.BufferedReader(IterStream(), buffer_size=buffer_size)
+
+def parse_excel_csv_reader(reader):
```

Comment:
This may want to be a `@classmethod` on the `ExcelParser`, just to keep everything in one place.

Reply:
This is also used in the CSV parser because most of the logic is shared with the Excel parser

Reply:
Right. Just wondering if logically we want to attach the function to one of the two classes and then call it in the other via `TheClass.the_method(...)`

Reply:
Ahh gotcha yeah I can move it!